docs/data/material/components/init-color-scheme-script/init-color-scheme-script.md

```diff
@@ -0,0 +1,140 @@
+---
+productId: material-ui
+title: InitColorSchemeScript component
+components: InitColorSchemeScript
+githubSource: packages/mui-material/src/InitColorSchemeScript
+---
+
+# InitColorSchemeScript
+
+<p class="description">InitColorSchemeScript component removes the dark mode flicker for server-side rendering application.</p>
+
+## Introduction
+
+`InitColorSchemeScript` component is used to remove the dark mode flicker for server-side rendering (SSR) applications.
+It is a script that runs before React to attach an attribute based on user preference.
+
+It is recommended to use this component whenever:
+
+- The application supports light and dark modes.
+- The application is server rendered (SSR).
+
+## Basics
+
+You need to enable CSS variables with `colorSchemeSelector: 'data'` in your theme.
+
+```js
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+const theme = createTheme({
+  cssVariables: {
+    colorSchemeSelector: 'data',
+  },
+});
+
+function App() {
+  return <ThemeProvider theme={theme}>{/* Your app */}</ThemeProvider>;
+}
+```
+
+Then, render the `InitColorSchemeScript` component as the first child of the `<body>` tag.
+
+Below are the location for rendering `InitColorSchemeScript` component for a specific framework.
+
+### Next.js App Router
+
+Place the `InitColorSchemeScript` component in the `layout.tsx` file:
```

Comment:
nitpick: the Next.js docs style these files with no extension (since the reader might be using JS or TS): `layout`, `_document`, etc.

```suggestion
Place the `InitColorSchemeScript` component in the root `layout` file:
```